### PR TITLE
Create tee utility

### PR DESCRIPTION
**Pull request by jeremyvii**
This PR begins implementing the `tee` utility. The `-i` flag has not been implemented, I was unsure how to best handle `SIGINT` and, unless I am misunderstanding the documentation, it seems that `write` ignores interrupts anyway. 

`-a`|`--append`

### CODE REVIEW

**Comment by jeremyvii** [comment]
I followed [this guide](https://rust-cli.github.io/book/tutorial/testing.html#testing-cli-applications-by-running-them) for the testing (which is why I am using the`assert_cmd` and `tempfile` crates). Although the tests are failing on various environments. I will look into these failures when I get a free moment.

File: tee/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+description = """
+Read from standard input and write to standard output or files.
+"""
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+clap = { version = "^2.33.0", features = ["wrap_help"] }
+assert_cmd = "1.0.2"
+tempfile = "3.1.0"
```
Comment (GrayJack) [inline tee/Cargo.toml]: These are only used in tests, right? The it should be in `dev-dependencies` instead, so we don't have to compile those crates unless we are testing it

**Review by GrayJack — COMMENTED** [review]
I didn't had the time to check the code behavior deeply, it looks good to me, but tomorrow I'll read more carefully and do some tests on my end.

About signal ignoring, I did ignore a signal in the `nohup` utility. It was the only usage of unsafe that I allowed in the tools for now (mostly because I didn't thought how to turn it in a better API). You can give a look there and at `man signal`.

File: tee/src/cli.rs [hunk 2]
```diff
@@ -18,4 +18,5 @@ pub(crate) fn create_app<'a, 'b>() -> App<'a, 'b> {
                 .short("a")
                 .long("append"),
         )
+        .arg(Arg::with_name("ignore").help("Ignore interrupt signals.").short("i").long("ignore"))
```
Comment (GrayJack) [inline tee/src/cli.rs]: I think the long name usually are called `ignore-interrupts`, but if you think that `ignore` is enough, I'm ok with that as long put `ignore-interrupts` as a visible alias

**Comment by jeremyvii** [comment]
I was able to resolve the test failures aside from the `cargo-deny` failure.

File: tee/src/cli.rs (hunk 2 above)
Comment (jeremyvii) [reply tee/src/cli.rs]: I went ahead and went with `ignore-interrupts`.

**Comment by GrayJack** [comment]
Thanks, lgtm, let's merge it

bors r+

**Pull request merged**